File: server/src/Makc2021.Layer4.Domains.DummyMain/Queries/List/Get/ListGetQueryDomainOutput.cs
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2021.Layer2.Queries.List.Get;
using Makc2021.Layer4.Domains.DummyMain.Queries.Item.Get;

namespace Makc2021.Layer4.Domains.DummyMain.Queries.List.Get
{
    /// <summary>
    /// Выходные данные запроса на получение списка в домене.
    /// </summary>
    public class ListGetQueryDomainOutput : ListGetQueryOutput<ItemGetQueryDomainOutput>
    {

    }
}

File: client/makc2021-react-redux-ts/src/Layer4/Domains/DummyMain/Queries/List/Get/ListGetQueryDomainOutput.ts
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

import {
  createListGetQueryOutput,
  ListGetQueryOutput
} from 'src/Layer2/Queries/List/Get/ListGetQueryOutput';
import { ItemGetQueryDomainOutput } from '../../Item/Get/ItemGetQueryDomainOutput';

/**
 * Выходные данные запроса на получение списка в домене.
 */
export interface ListGetQueryDomainOutput
  extends ListGetQueryOutput<ItemGetQueryDomainOutput> {}

/**
 * Создать выходные данные запроса на получение списка в домене.
 * @returns Выходные данные запроса на получение списка в домене.
 */
export function createListGetQueryDomainOutput() {
  return {
    ...createListGetQueryOutput<ItemGetQueryDomainOutput>()
  } as ListGetQueryDomainOutput;
}
